src/ethereum/london/vm/gas.py

```diff
@@ -159,10 +174,10 @@ def calculate_call_gas_cost(
     call_gas_cost: `ethereum.base_types.Uint`
         The total gas amount for executing Opcodes `CALL` and `CALLCODE`.
     """
-    if gas_left < extra_gas:
-        raise OutOfGasError
+    if gas_left < extra_gas + memory_cost:
+        return gas + extra_gas
 
-    gas = min(gas, max_message_call_gas(gas_left - extra_gas))
+    gas = min(gas, max_message_call_gas(gas_left - extra_gas - memory_cost))
```

Comment:
Looking at this again, I think the new version is less understandable. Could you walk me through what's going on here and why we're making this particular change?

Reply:
The main idea was that we want to to be able to deduct gas in one single go (inside `charge_gas`) in order to be able to implement the proper `evm_trace`. For that, we need to calculate the `memory_cost` but not deduct it from the `evm.gas_left` right away (which is what we currently). This step is to account for that fact i.e. the `gas_left` is effectively `gas_left - memory_cost`

Reply:
Ok... that makes sense. So what about the line that used to `raise OutOfGasError`?

Reply:
`OutOgGasError` will now be called in the `charge_gas` function

Reply:
Also, I will change `if gas_left < extra_gas + memory_cost:` to `if gas_left - memory_cost < extra_gas:` which I think makes it a bit clearer